tools/ci_build/provider_ops_disabler.py

```diff
@@ -0,0 +1,334 @@
+
+        elif onnx_versioned_typed_op in code_line:
+            # e.g. class ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(
+            #          kCpuExecutionProvider, kOnnxDomain, 7, 9, float, Upsample);
+            # e.g. BuildKernelCreateInfo<ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(
+            #          kCpuExecutionProvider, kOnnxDomain, 1, 10, float, LogSoftmax)>,
+            trim_at = code_line.index(onnx_versioned_typed_op) + onnx_versioned_typed_op_len
+            *_, domain, opset_from, opset_to, _, op_type =\
+                [arg.strip() for arg in code_line[trim_at: -len(end_mark)].split(',')]
+            call_back_ret = call_back(op_type, opset_from, opset_to, domain)
+
+        return offset + 1, call_back_ret  # end of process_lines(...)
+
+    lines = []
+    with open(ep_path, 'r') as file_to_read:
+        lines = file_to_read.readlines()
+
+    backup_path = ep_path + '~'
+    if not os.path.isfile(backup_path):
+        shutil.move(ep_path, backup_path)
+
+    with open(ep_path, 'w') as file_to_write:
+        line_offset = 0
+
+        while line_offset < len(lines):
+
+            line = lines[line_offset]
+            stripped = line.strip()
+
+            if stripped.startswith('class ONNX_OPERATOR'):
+                # collect op versions
+
+                next_line_offset, _ = process_lines(lines,
+                                                    line_offset,
+                                                    ');',
+                                                    fill_version_map)
+
+                for index in range(line_offset, next_line_offset):
+                    file_to_write.write(lines[index])  # leave as it was
+
+                line_offset = next_line_offset
+
+            elif stripped.startswith('BuildKernelCreateInfo<ONNX'):
+                # comment out unused ops
+
+                next_line_offset, disabled = process_lines(lines,
+                                                           line_offset,
+                                                           ')>,',
+                                                           disable_op)
+
+                for index in range(line_offset, next_line_offset):
+                    if disabled:  # comment out unused
+                        file_to_write.write('//' + lines[index])
+
+                    else:  # leave as it is
+                        file_to_write.write(lines[index])
+
+                line_offset = next_line_offset
+
+            else:  # leave as it is
+                file_to_write.write(line)
+                line_offset += 1
+
+    # end of rewrite_cpu_provider(...)
+
+
+if __name__ == "__main__":
+
+    PARSER = argparse.ArgumentParser(
+        description="provider rewriter",
+        usage="""
+        --model_path <path to model(s) folder>
+        --file_path <path to file of ops>
+        --ep_path <path to provider file>
```

Comment:
Help could be clearer. Some could consider the 'file of ops' to be the registrations, so maybe be more specific and say config file with ops to enable. Would be nice to have a description of the format somewhere in this help text so it's directly available, or alternatively a link to the help document.

Reply:
Sure, adding the description.

Reply:
Done.